### ISSUE TITLE
Pin on compatible version of gdal

### ISSUE THREAD

**Issue by SylvainCorlay** [conda-forge/fiona-feedstock#117]
Fiona requires any version of gdal at runtime independently of the one that was used to build it.

Maybe having a `pin_compatible` statement in the run section instead of a wild card dependency on gdal would be better.

Also gdal misses a `run_exports` section.

cc @jorisvandenbossche

**Comment by ocefpaf** [comment]
> Fiona requires any version of gdal at runtime independently of the one that was used to build it.

That is not true, we have a global pinning to [`gdal 2.4`](https://github.com/conda-forge/fiona-feedstock/blob/master/.ci_support/osx_python3.7.yaml#L11-L12), but rebuilds of that version without a rebuild of `fiona` may cause trouble. (Not sure if `pin_compatible` would fix that though.)

> Also `gdal` misses a run_exports section.

`gdal` are the python bindings and I'm not sure we need it, [`libgdal` does have it](https://github.com/conda-forge/gdal-feedstock/blob/master/recipe/meta.yaml#L66-L69).

Can you describe the problems you are facing? Maybe it is related to the old label packages still lingering on `main`.

**Comment by ocefpaf** [comment]
Yeah, the `geopandas` install should have triggered a `libgdal` update. Maybe we have a package with bad metadata or maybe `conda` is at fault. I'll try to investigate.

**Comment by parente** [comment]
> Maybe we have a package with bad metadata or maybe conda is at fault. I'll try to investigate.

@ocefpaf FWIW, in an environment we build with many packages, conda 4.5.12 happily installed the latest geopandas and fiona builds, but kept gdal at 2.3.2 which broke fiona with a missing symbol error.. I haven't had a chance to run an ablation experiment to track down what other package might be the culprit but it feels like there's a package with bad metadata somewhere.
